src/Stratis.Bitcoin.IntegrationTests/StratisMinerTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using NBitcoin;
+using Stratis.Bitcoin.Features.BlockStore;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.Miner;
+using Stratis.Bitcoin.Features.Notifications;
+using Stratis.Bitcoin.Features.RPC;
+using Stratis.Bitcoin.Features.Wallet;
+using Stratis.Bitcoin.Features.Wallet.Interfaces;
+using Stratis.Bitcoin.Features.WatchOnlyWallet;
+using Xunit;
+
+namespace Stratis.Bitcoin.IntegrationTests
+{
+    public class StratisMinerTests
+    {
+        [Fact]
+        public void CanMineStratisCoins()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create())
+            {
+                CoreNode node1 = builder.CreateStratisPosNode(true, fullNodeBuilder =>
+                {
+                    fullNodeBuilder
+                        .UseStratisConsensus()
+                        .UseBlockStore()
+                        .UseMempool()
+                        .UseWallet()
+                        .AddPowPosMining()
+                        .AddRPC();
+                });
+
+                node1.NotInIBD();
```

Comment:
NotInIBD(); will try to get a full node reference. This will fail at this moment because you need to call ' builder.StartAll();' that will assign it.

Reply:
I am using a somewhat different construct to the other tests, in this case the node gets started as soon as it is built. Note the 'true' argument to CreateStratisPosNode.